### ISSUE TITLE
prohibited_super_call rule false violation with NSFileProviderExtension providePlaceholder override

### ISSUE THREAD

**Issue by BenStaveleyTaylor** [realm/SwiftLint#2212]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

The prohibited_super_call rule generates an incorrect violation report when linting the `providePlaceholder` override for an `NSFileProviderExtension` implementation. The code to repro shown below is taken directly from the Apple template code generated for a new FileProvider extension target and does not call super.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --config swiftlint.yml 
Loading configuration from 'swiftlint.yml'
Linting Swift files in current working directory
Linting 'FileProviderExtension.swift' (1/1)
/Users/bestave/Desktop/DeleteMe/DeleteMeFP/ProhibitedSuperTest/FileProviderExtension.swift:5:99: warning: Prohibited calls to super Violation: Method 'providePlaceholder(at:completionHandler:)' should not call to super function (prohibited_super_call)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.25.1 (release and latest source)
* Installation method used: SwiftLint.pkg installer and also cloned/built from master source
* Paste your configuration file:

```yml
opt_in_rules:
  - prohibited_super_call
```

* Are you using [nested configurations]? No
* Which Xcode version are you using? 9.3.1
* Do you have a sample that shows the issue? Yes:

```swift
import FileProvider

public class FileProviderExtension: NSFileProviderExtension {

    override func providePlaceholder(at url: URL, completionHandler: @escaping (Error?) -> Void) {
        guard let identifier = persistentIdentifierForItem(at: url) else {
            completionHandler(NSFileProviderError(.noSuchItem))
            return
        }
    }

}
```
